File: LoanManagement.Entities/Enums/Enums.cs
namespace LoanManagement.Entities.Enums;

public enum LoanType
{
    Personal = 0, //İhtiyaç Kredisi
    Education = 1,  //Eğitim Kredisi
    Vehicle = 2     //Taşıt Kredisi
}

public enum LoanStatus
{
    Active = 1, //Aktif (ödemeleri devam ediyor.)
    Closed = 2  //Kapatıldı (Tüm borçlar bitti.)
}

public enum InstallmentStatus
{
    Unpaid = 1, //Ödenmedi
    Paid = 2,   //Ödendi
    Overdue = 3, //Gecikmiş(Vadesi geçti ama ödenmedi)
}

File: frontend/src/types/enums.ts
// .NET enum sözleşmesi
// Request tarafında numeric int kullanılır; response'larda backend lokalize string döner.

export enum LoanType {
  Personal = 0, // İhtiyaç Kredisi
  Education = 1, // Eğitim Kredisi
  Vehicle = 2, // Araç / Taşıt Kredisi
}

export const LoanTypeLabels: Record<LoanType, string> = {
  [LoanType.Personal]: "İhtiyaç Kredisi",
  [LoanType.Education]: "Eğitim Kredisi",
  [LoanType.Vehicle]: "Araç Kredisi",
};

export type LoanStatus = "Active" | "Closed";

export const LoanStatusLabels: Record<LoanStatus, string> = {
  Active: "Aktif",
  Closed: "Kapatıldı",
};

export type InstallmentStatus = "Unpaid" | "Paid" | "Overdue";

export const InstallmentStatusLabels: Record<InstallmentStatus, string> = {
  Unpaid: "Ödenmedi",
  Paid: "Ödendi",
  Overdue: "Gecikmiş",
};

export type AppRole = "Admin" | "Customer";
